web/client/epics/wfsquery.js

```diff
@@ -178,9 +178,8 @@ export const wfsQueryEpic = (action$, store) =>
             // getSelected Layer and merge layerFilter and cql_filter in params  with action filter
             const layer = selectedLayerSelector(store.getState());
 
-            const {layerFilter, params} = layer ?? {};
-            const cqlFilter = find(Object.keys(params || {}), (k = "") => k.toLowerCase() === "cql_filter");
-
+            const {layerFilter, params = {}} = layer ?? {};
+            const cqlFilter = params?.[find(Object.keys(params || {}), (k = "") => k.toLowerCase() === "cql_filter")];
```

Comment:
why the need of k = "" , do we have keys null or undefined?

Reply:
also can you add unit tests here, i'm 100% convinced about this refactor by only looking at the code

Reply:
Not a refactor but a fix. In fact the original was not working (taking the key "cql_filter", not the value). 

Need to add a test for this.

Reply:
added a unit test :+1: